### ISSUE TITLE
`docker import` does not honor the `--platform` flag

### ISSUE THREAD

**Issue by jodersky** [moby/moby#42813]
**Description**

When creating an image with the `docker import` command, the `--platform` flag appears to be ignored, and the host's architecture is always used.

**Steps to reproduce the issue:**

1. Create a dummy root file system:
    ```
    mkdir rootfs && touch rootfs/a
    ```
2. On an amd64 host, create an image containing above filesystem, but override the platform to something else, such as `linux/armv7`:
   ```
   tar -C rootfs -c . | docker image import --platform linux/armv7 -
   ```
3. Inspect the image:
   ```
   docker inspect <sha256 of generated image> | jq .[].Architecture
   ```

**Describe the results you received:**

The image's architecture will be set to the host's architecture, "amd64" in this case.

**Describe the results you expected:**

The image's architecture should be set to what was given via the `--platform` flag, "armv7" in this case.

**Additional information you deem important (e.g. issue happens only occasionally):**

This issue does not appear when creating the image via an equivalent `docker build`  (BUILDKIT enabled). E.g., consider the equivalent Dockerfile and build command:

```
FROM scratch
COPY rootfs /
```
```
DOCKER_BUILDKIT=1 docker build . --platform linux/armv7
```

In this case, the architecture of the resulting image is correctly set to armv7.

**Output of `docker version`:**

```
Client:
 Version:           20.10.5+dfsg1
 API version:       1.41
 Go version:        go1.15.9
 Git commit:        55c4c88
 Built:             Wed Aug  4 19:55:57 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.5+dfsg1
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.15.9
  Git commit:       363e9a8
  Built:            Wed Aug  4 19:55:57 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.5~ds1
  GitCommit:        1.4.5~ds1-2
 runc:
  Version:          1.0.0~rc93+ds1
  GitCommit:        1.0.0~rc93+ds1-5+b2
 docker-init:
  Version:          0.19.0
  GitCommit:
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 30
  Running: 4
  Paused: 0
  Stopped: 26
 Images: 121
 Server Version: 20.10.5+dfsg1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 1.4.5~ds1-2
 runc version: 1.0.0~rc93+ds1-5+b2
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.0-8-amd64
 Operating System: Debian GNU/Linux 11 (bullseye)
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.4GiB
 Name: jodersky-x1
 ID: DRDM:7KLF:RWSD:C2NW:MCOE:GCIX:YMPF:YBRO:JICK:ORKL:MJSD:TFI6
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 79
  Goroutines: 90
  System Time: 2021-09-03T09:02:41.271149459+02:00
  EventsListeners: 0
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: Support for cgroup v2 is experimental
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

x86_64 laptop running Debian Bullseye (the current stable version).

**Comment by simonsd** [comment]
I can confirm this is still an issue

Output of docker version
`Client:
 Cloud integration: 1.0.17
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:55:20 2021
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:53:48 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0`

**Comment by thaJeztah** [comment]
Thanks for reporting.

I did a quick search in the source code, and I think I see why this is the case. From the source code, I see that the `--platform` flag on `docker image import` was added in https://github.com/moby/moby/commit/0380fbff37922cadf294851b1546f4c212c7f364#diff-2d8d96f6fa256de94f3737c8631e99d897efb15ec869f465f8cfcb8962e70be0R116 (as part of https://github.com/moby/moby/pull/34642). In that PR, it was added to support the experimental LCOW (Linux Containers On Windows) feature, and to allow importing a Linux image on a Windows daemon.
Given that (at the time) it was only to support that specific case, the "architecture" information was discarded, and only the Operating System (windows / linux) was used.

A later commit https://github.com/moby/moby/commit/337ba71fc1124603302e28d94e2f08674e31a756 (part of https://github.com/moby/moby/pull/37350) changed _some_ of the code to take the full platform information into account (for `docker build` and `docker pull`), but did not yet update the `docker image import`.

I _think_ it should be possible to pass through the whole platform information, but not likely for a patch release for the current (20.10.x) release.
